src/components/ButtonWithDropdownMenu/types.ts

```diff
@@ -132,6 +132,9 @@ type ButtonWithDropdownMenuProps<TValueType> = {
     /** The second line text displays under the first line */
     secondLineText?: string;
 
+    /** Callback to execute when a dropdown submenu option is selected */
+    onSubitemSelected?: (selectedItem: PopoverMenuItem, index: number, event?: GestureResponderEvent | KeyboardEvent) => void;
```

Comment:
Let's use camelCase to be consistent:

```suggestion
    onSubItemSelected?: (selectedItem: PopoverMenuItem, index: number, event?: GestureResponderEvent | KeyboardEvent) => void;
```